### PR TITLE
Infinite Scrolling added

### PR DESCRIPTION
**Pull request by vishalrishu**
- Infinite Scrolling Added in home page
- Edit/Delete option added for a post

### CODE REVIEW

**Comment by vishalrishu** [comment]
insta-app

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -13,13 +13,12 @@
       </div>
       <%= link_to @post.user.username, user_path(@post.user), class: 'username' %>
     </div>
-    <div class="description">
-      <%= render partial: "description", locals: {post: @post} %>
-    </div>
+    <%= @post.description %>
     <div class="delete">
       <% if current_user.posts.exists?(@post.id) %>
```
Comment (jaindheerain) [inline app/views/posts/show.html.erb:18]: this line should not be there in view, views should be as dumb as possible. This line queries to db to check .

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,23 +1,42 @@
...
+    @post = current_user.posts.find(params[:id])
+    @post.destroy
+
+    redirect_to root_path
+  end
+
+  private
+
+  def set_post
+    @post  = Post.find(params[:id])
```
Comment (jaindheerain) [inline app/controllers/posts_controller.rb:37]: what happens if there is not post with id passed in params,

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,23 +1,23 @@
 class UsersController < ApplicationController
-    before_action :authenticate_user!, only: [:edit, :update]
+  before_action :authenticate_user!, only: [:edit, :update]
 
-    def show
-        @user  = User.find(params[:id])
-        @posts = @user.posts.order(created_at: :desc)
-    end
+  def show
+    @user  = User.find(params[:id])
```
Comment (jaindheerain) [inline app/controllers/users_controller.rb:5]: shift these queries to model, a single function in user that returns all the posts from the model.
something like. User.get_all_posts(user_id), keep most of the DB interactions in model.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
+    @posts = @user.posts.order(created_at: :desc)
```
Comment (jaindheerain) [inline app/controllers/users_controller.rb:6]: shift these queries to model, a single function in user that returns all the posts from the model.
something like. User.get_all_posts(user_id), keep most of the DB interactions in model.
Both lines 5,6 can be shifted to the model.

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1,4 +1,11 @@
...
+  <div class='form-upload'>
+    <%= render 'upload_form' %>
+  </div>
+
+  <div class="posts">
+    <% @posts.each do |post| %>
+      <section class="post">
   <div class="user">
     <div class="avatar">
       <% if post.user.avatar.attached? %>
```
Comment (jaindheerain) [inline app/views/posts/index.html.erb:11]: this too queries db in the view.

Comment (vishalrishu) [reply app/views/posts/index.html.erb:11]: Thanks. Will update that

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (vishalrishu) [reply app/controllers/posts_controller.rb:37]: For this, I changed find to find_by. Now, I'm redirecting to root with some error msg for this case

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (vishalrishu) [reply app/controllers/users_controller.rb:6]: Done

**Pull request merged**